### DISCUSSION TITLE
Поддержка раскладки без перевода подсказок

### DISCUSSION THREAD

**Discussion by py-art** [Wansmer/langmapper.nvim#16] [General]
Здравствуйте Иван!

Подскажите, есть ли возможность отключения перевода английских букв и сочетаний клавиш при отображении в подсказках?

Очень мешает когда во всех подсказках русские буквы!!! Я привык к английским. И сочетания клавиш набираю в слепую, не глядя на клавиатуру.

Идеально было бы, если просто при переключении на другую раскладку всё работало так, как будто ты на английской раскладке сидишь, но при этом всё, ну или почти всё,  работает точно так же! Хотя бы чтоб работало в нормальном режиме и в режиме выделения.

Из-за не знания как отключить такое поведение, я попробовал ваш плагин несколько месяцев назад, понял, что это не то что я хочу, и перестал пользоваться.

Спустя несколько месяцев меня начало напрягать постоянное переключение раскладки, хотя langmap у меня настроен. И я опять забрёл через поисковик на Ваш плагин.

Если такого функционала в данный момент нет, то пару вопросов:
1. Предполагается ли в дальнейшем?
2. Если Да, то когда?

Если такое уже существует в данной версии, то как настроить применимо к LazyVim ???

**Comment by Wansmer** [comment]
Здравствуйте!

Не понял, о каких подсказках идет речь. Если про функционал плагинов, вроде `which-key.nvim`, то нужно смотреть, есть ли у них возможность фильтрации для вывода подсказок, и если есть, то фильтровать по русским буквам. Т.к. этот функционал не относится к langmapper.nvim, то и реализован он не будет.

> Идеально было бы, если просто при переключении на другую раскладку всё работало так, как будто ты на английской раскладке сидишь, но при этом всё, ну или почти всё, работает точно так же! Хотя бы чтоб работало в нормальном режиме и в режиме выделения.

Если правильно настроены layout и default_layout, то именно так плагин и работает. Подробности настройки в readme. Для более точного понимания работы плагина - есть статься на [хабре](https://habr.com/ru/articles/726400/).

**Comment by py-art** [comment]
Да, я именно по which-key.nvim говорю, остальные плагины не смотрел при запуске вашего плагина. При включении вашего плагина происходит замена в подсказках английской "q" на русскую "й" и т.п.
Можно это отключить как-то?

Статью на хабре читал. С неё и пришёл на эту репу и в первый раз и в этот раз. Хабр - рулит))

Документации для "дебилов" немного не хватает! ))))
Что-то типа:
- хочешь это - пропиши "вот такое" в настройках подключения плагина
- хочешь другое - пропиши "вот так"

А в отдельном файле на github прописать, что каждая настройка делает.

Я так и не понял, как отключить функцию, чтоб ваш плагин не переводил в подсказках сочетания клавиш с английского на русский. Но чтоб при этом команды в русской раскладке выполнялись так же как и в английской. 
Не могли бы вы пример скинуть, как это сделать?

**Reply by Wansmer** [reply]
Тут проблема в том, что `which-key` сам собирает все переведенные сочетания, т.е. я из плагина никак не другой плагин повлиять не могу. Сам я `which-key` не пользуюсь, но уверен, что там должна быть настройка для фильтрации подсказок, как писал выше. Попробуйте поискать в документации или в ишьюс `which-key`. Если такое есть, то просто фильтруйте по условию на отсутствие русских букв.

**Comment by py-art** [comment]
Понял.
Спасибо за мгновенную связь!
Я поковыряюсь. Возможно вам PR отправлю, если сам разберусь как это сделать.

**Reply by py-art** [reply]
А нормальной документации к плагину всё же не хватает!!!
При наличии документации плагин развивался бы лучше.
Я считаю - это крутой плагин!

**Reply by Wansmer** [reply]
Лучше не PR, а пост в специальной теме по взаимодействию с другими плагинами: https://github.com/Wansmer/langmapper.nvim/discussions/11
